tools/targets/STM32_gen_PeripheralPins.py

```
@@ -979,6 +937,141 @@ def print_usb(lst):
         out_c_file.write( "#endif\n" )
 
 
+def print_PinList(l):
+    l.sort(key=natural_sortkey)
+    previous_pin = ""
+    for p in l:
+        print_debug("pin %s => %s" % (p, p[0]))
+        if p[0] == previous_pin:
+            continue
+        previous_pin = p[0]
+
+        if "_ALT" in p[0]:
+            s1 = "    %-10s = %s | %s, // same pin used for alternate HW\n" % (p[0], p[0].split('_A')[0], p[0].split('_')[2])
+        else:
+            if "PA" in p[0]:
+                PinValue = 0
+            elif "PB" in p[0]:
+                PinValue = 0x10
+            elif "PC" in p[0]:
+                PinValue = 0x20
+            elif "PD" in p[0]:
+                PinValue = 0x30
+            elif "PE" in p[0]:
+                PinValue = 0x40
+            elif "PF" in p[0]:
+                PinValue = 0x50
+            elif "PG" in p[0]:
+                PinValue = 0x60
+            elif "PH" in p[0]:
+                PinValue = 0x70
+            elif "PI" in p[0]:
+                PinValue = 0x80
+            elif "PJ" in p[0]:
+                PinValue = 0x90
+            elif "PK" in p[0]:
+                PinValue = 0xA0
+            elif "PZ" in p[0]:
+                PinValue = 0x0 # to update
+            else:
+                print("error in print_PinList with pin %s" % p[0])
```

Do we want to continue when this error occurs? Should we raise an exception and/or use `logging.error` instead of `print`?